### DISCUSSION TITLE
Design for onboarding / first run setup tool

### DISCUSSION THREAD

**Discussion by mairin** [podman-desktop/podman-desktop#3244] [Design Feedback]
Hi 👋🏻 

I want to share some mockups from a feature we've been planning called the **Onboarding** feature. The feature is aimed at first-time users of Podman Desktop, and the idea is to give an optional guided / step-by-step flow at setting up the tool. It would present a list of Podman Desktop extensions you can choose to set up (see upper right corner on mockups below past the first screen), and based on the extensions you selected, it would walk you through setting them up in order:

![Screenshot from 2023-07-17 16-07-05](https://github.com/containers/podman-desktop/assets/799683/f28a9e38-4398-4fb5-9654-3e1e208af38e)

_**Note:** there's a ton of ways to skip out of this if you don't want to do it - in the upper left, hitting cancel in the lower right, and Xing the window (the X isn't in this mockup because I'm lazy, but it'll be there :) )_

Let's say you choose to go through this and want to walk through setting up Podman. The Podman extension relies on having (depending on your platform) a podman machine set up or a podman installed natively on the OS (Linux) - so first step is to check if you have Podman on your system or not. If you don't:

![Screenshot from 2023-07-17 16-09-51](https://github.com/containers/podman-desktop/assets/799683/9bc26461-03bb-4438-8b5c-575dd9d68c70)
 
So in this case we'll go ahead and install it for you:

![Screenshot from 2023-07-17 16-12-44](https://github.com/containers/podman-desktop/assets/799683/40bc9139-0f2c-4546-aa49-d3b00fcfc9d7)

After a bit, it'll be set up, and we'll let you know it worked, and offer some options (which you can always revisit in Podman Desktop's Settings area):

![Screenshot from 2023-07-17 16-13-24](https://github.com/containers/podman-desktop/assets/799683/33550966-6c40-4f43-a0e2-71ce0c215719)

We have some designs for this onboarding setup for a number of the extensions shown in the mockups; I'm just showing the simplest Podman-specific ones to demonstrate the basic idea of the feature. Does this make sense? Is this something you would find useful to set up Podman Desktop or to help someone new to Podman Desktop get started with us? What kinds of set up would you like help with via  tool like this?

We would love to hear your feedback :) And I'm happy to post more mockups showing more bits of this for the curious.

**Comment by afbjorklund** [comment]
> So in this case we'll go ahead and install it for you

I'm curious how this would actually work on Linux, if it would just be "greyed out" in case you are not on Fedora...

Or will it just do a `sudo apt install podman` and hope for the best ? As in [packages.ubuntu.com/podman](https://packages.ubuntu.com/search?keywords=podman&searchon=names&exact=1&suite=all&section=all) (v3.4.4)

Because is there is no installer, only `podman-remote`

**Reply by mairin** [reply]
Hey @afbjorklund - that's a good point. I'm very Fedora/RHEL centric for obvious reasons (😅) so I had been thinking when mocking this up that we'd be doing a dnf install in the background if it wasn't present... @benoitf do you have any ideas on how we could handle Podman install on Linux in a distro-agnostic way? 😖

**Reply by benoitf** [reply]
we have ways to detect the underlying Operating System.
we could support certain Operating System and commands to execute ( for example vs code remote is doing this kind of stuff to install something once connected to a container)

But we could also just give pointers to a documentation when we don't support or if pre-requisites are not there ( for example need root access but user has not this privilege, unsupported distro, too old distro.)

I would suggest for now instructions and then iterate on automation for some platforms. As there are a lot of stuff to maintain and only few users are Linux users for now.

**Reply by afbjorklund** [reply]
Previously I was involved a bit in the PackageKit project, that tried to handle the privilege escalation and everything that was involved with clicking a web link to install a package. Not sure if that project is still feasible.

https://www.freedesktop.org/software/PackageKit/

Anyway, it just ran a root daemon (packagekitd) so I guess that goes against everything Podman stands for. The "wall of text" is probably the best option, for now. https://podman.io/docs/installation#installing-on-linux

**Reply by mairin** [reply]
@afbjorklund smtg like PolicyKit?

**Reply by afbjorklund** [reply]
The other container runtimes just provides a script that you run, to detect and configure your package manager.

`curl -fsSL https://get.docker.com | sh`

**Comment by loiclacombe** [comment]
This would be really nice. As a technical lead, I often have to write documentation for our teams, for the onboarding process, and walking people through a first run wizard is really easier than documenting navigation through panels or menus. I am all for it.

**Reply by mairin** [reply]
Love to hear it, @loiclacombe ! 🦭😃

**Comment by cedricclyburn** [comment]
Ah this would be amazing @mairin! I just encountered some friction getting Podman Desktop setup on Windows #3421, this would be a great help :)

**Comment by rujutashinde** [comment]
closing this discussion as it is either implemented or not needed
